collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -196,18 +206,37 @@ public void onClick(View v) {
     @Override
     protected void onResume() {
         super.onResume();
-        viewModel.resume();
+        mainMenuViewModel.resume();
 
         setButtonsVisibility();
         invalidateOptionsMenu();
     }
 
     private void setButtonsVisibility() {
-        reviewDataButton.setVisibility(viewModel.shouldEditSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        sendDataButton.setVisibility(viewModel.shouldSendFinalizedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        viewSentFormsButton.setVisibility(viewModel.shouldViewSentFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        getFormsButton.setVisibility(viewModel.shouldGetBlankFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        manageFilesButton.setVisibility(viewModel.shouldDeleteSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        reviewDataButton.setVisibility(mainMenuViewModel.shouldEditSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        sendDataButton.setVisibility(mainMenuViewModel.shouldSendFinalizedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        viewSentFormsButton.setVisibility(mainMenuViewModel.shouldViewSentFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        getFormsButton.setVisibility(mainMenuViewModel.shouldGetBlankFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        manageFilesButton.setVisibility(mainMenuViewModel.shouldDeleteSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        final MenuItem projectsMenuItem = menu.findItem(R.id.projects);
+
+        ProjectIconView projectIconView = (ProjectIconView) projectsMenuItem.getActionView();
+        projectIconView.setProject(currentProjectViewModel.getCurrentProject().getValue().getValue());
+        projectIconView.setOnClickListener(v -> onOptionsItemSelected(projectsMenuItem));
+
+        currentProjectViewModel.getCurrentProject().getValue().consume();
+        currentProjectViewModel.getCurrentProject().observe(this, project -> {
```

Comment:
The `observe` can just happen in `onCreate` I think. The current project could either be stored in a field in the Activity and then used in `onPrepareOptionsMenu` (as it gets called after `invalidateOptionsMenu`), or you could just call `currentProjectViewModel.getCurrentProject().getValue()` in `onPrepareOptionsMenu`.

Reply:
> or you could just call currentProjectViewModel.getCurrentProject().getValue() in onPrepareOptionsMenu

I do that. The reason why I put it here is to avoid performing what's between those brackets when the activity is created. Once it's called to setup the icon (a few lines above) it's marked as consumed and not used anymore.

Reply:
Yeah sorry this and the comment on `Consumable` should be one really as they are very related (as you point out).

Reply:
Fixed.